tests/PHPStan/Analyser/data/bug-10076.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Bug10076;
+
+/**
+ * @template T
+ *
+ * @param 'object'|'array'|class-string<T> $type
+ *
+ * @return list<mixed>
+ */
+function result($type = 'object'): array
+{
+	return [];
+}
+
+result();
+result('object');
+result('array');
+result(\DateTime::class);
```

Comment:
Please do not mix function declarations and side-effects such as function calls. Please put these calls in an anonymous function.